### PR TITLE
Preparations towards a reproducible release build

### PR DESCRIPTION
**Pull request by kaikreuzer**
- using an ESH release build (0.10.0.b1)
- added ESH release p2 repos for release builds
- removed targetplatform, which afaics isn't used at all anymore

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@pfink FYI: Just to be safe, I switched the master to an ESH release build and also made sure that we have a p2 repo referenced, which will stay (so far we only listed the esh stable p2 repo, which is replaced on every new stable build - which causes to non-reproducible release builds, which is bad (see https://github.com/openhab/openhab-distro/issues/529)).

File: poms/tycho/pom.xml [hunk 1]
```diff
@@ -214,11 +214,31 @@
...
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+        </repository>
+
+        <!-- SmartHome p2 release repository -->
+        <repository>
+            <id>p2-smarthome-release</id>
+            <url>http://download.eclipse.org/smarthome/updates-release/${esh.version}</url>
```
Comment (kaikreuzer) [inline poms/tycho/pom.xml]: Hm, this causes now an issue on the pipeline:
```
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-smarthome-release' from location http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT: No repository found at http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT. -> [Help 1]
```
@pfink Any clever idea how to handle this best? It is correct that this repo does not exist for esh snapshots, so IF esh.version contains "-SNAPSHOT", we must not use this repository.

Comment (pfink) [reply poms/tycho/pom.xml]: Why is this repo then enabled for snapshots? https://github.com/openhab/openhab-core/blob/master/poms/tycho/pom.xml#L240

Comment (kaikreuzer) [reply poms/tycho/pom.xml]: It is enabled for snapshot builds and this is what we are doing here: An openHAB 2.2.0-SNAPSHOT build, which uses an ESH release version, so we need access to this repo.

Comment (pfink) [reply poms/tycho/pom.xml]: I think you misunderstood this configuration option: It does refer to the artifact that should be downloaded and not to the artifact that is currently built.

Comment (kaikreuzer) [reply poms/tycho/pom.xml]: For a test I did 
```
        <!-- SmartHome p2 release repository -->
        <repository>
            <id>p2-smarthome-release</id>
            <url>http://download.eclipse.org/smarthome/updates-release/${esh.version}</url>
            <layout>p2</layout>
            <releases>
                <enabled>true</enabled>
                <updatePolicy>never</updatePolicy>
            </releases>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
        </repository>
```
and set esh.version=0.10.0-SNAPSHOT.
The situation isn't any better:
```
[INFO] Adding repository https://openhab.jfrog.io/openhab/eclipse-smarthome-stable
[INFO] Adding repository http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT
[ERROR] Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-smarthome-release' from location http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT: No repository found at http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT. -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Failed to load p2 repository with ID 'p2-smarthome-release' from location http://download.eclipse.org/smarthome/updates-release/0.10.0-SNAPSHOT
```

How can we solve this best? Shall we manually remove this repo from the pom, whenever we refer to esh snapshots and add it again, if we switch to a release? But that would need to be done in the pipeline as well (as this is where it currently breaks).

Comment (pfink) [reply poms/tycho/pom.xml]: Mhm, it seems like it still wants to resolve the repository url even though all artifacts can be grabbed from other repositories... Two possible solutions I see:
* Do not use esh.version as a property, use a separate property (e.g. `esh.releaseVersion`)
* Put it into a profile that is only used during the release build like here: https://github.com/openhab/openhab-distro/blob/master/pom.xml#L215

Comment (kaikreuzer) [reply poms/tycho/pom.xml]: Which solution would you prefer?

Comment (pfink) [reply poms/tycho/pom.xml]: They're both not perfect, but I'd like the first one more as it does not require passing extra arguments to the release build to make it work.

Comment (pfink) [reply poms/tycho/pom.xml]: Btw: Even though it was not the final solution, this repo should still be disabled for snapshots.

**Pull request merged**